File: EticaretJava/app/src/main/java/com/example/eticaretjava/repo/RetrofitProvider.java
package com.example.eticaretjava.repo;

import com.example.eticaretjava.service.AuthApi;
import com.example.eticaretjava.service.ApiService;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitProvider {

    private static final String BASE_URL = "https://alperensaracdeneme.com/eticaret/api/"; // sondaki / şart

    private static final HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();

    static {
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
    }

    private static final OkHttpClient okHttpClient = new OkHttpClient.Builder()
            .addInterceptor(loggingInterceptor)
            .build();

    public static final Retrofit retrofit = new Retrofit.Builder()
            .baseUrl(BASE_URL)
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    public static final ApiService.ProductApi productApi = retrofit.create(ApiService.ProductApi.class);
    public static final AuthApi authApi = retrofit.create(AuthApi.class);
    public static final ApiService.CartApi cartApi = retrofit.create(ApiService.CartApi.class);
    public static final ApiService.OrderApi orderApi = retrofit.create(ApiService.OrderApi.class);
}

File: Eticaretjetpack/app/src/main/java/com/example/eticaretjetpack/service/RetrofitProvider.kt
package com.example.eticaretjetpack.repo

import com.example.eticaretjetpack.service.AuthApi
import com.example.eticaretjetpack.service.ProductApi
import com.example.eticaretjetpack.service.CartApi
import com.example.eticaretjetpack.service.OrderApi
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitProvider {
    private const val BASE_URL = "https://alperensaracdeneme.com/eticaret/api/" // sondaki / şart

    private val loggingInterceptor = HttpLoggingInterceptor().apply {
        level = HttpLoggingInterceptor.Level.BODY
    }

    private val okHttpClient: OkHttpClient = OkHttpClient.Builder()
        .addInterceptor(loggingInterceptor)
        .build()

    val retrofit: Retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .client(okHttpClient)
        .addConverterFactory(GsonConverterFactory.create())
        .build()

    val productApi: ProductApi = retrofit.create(ProductApi::class.java)
    val authApi: AuthApi = retrofit.create(AuthApi::class.java)
    val cartApi: CartApi = retrofit.create(CartApi::class.java)
    val orderApi: OrderApi = retrofit.create(OrderApi::class.java)
}
